File: Whishpermate/WhisperMateShared/Models/Shortcut.swift
import Foundation

public struct Shortcut: Identifiable, Codable {
    public let id: UUID
    public var voiceTrigger: String
    public var expansion: String
    public var isEnabled: Bool

    public init(id: UUID = UUID(), voiceTrigger: String, expansion: String, isEnabled: Bool = true) {
        self.id = id
        self.voiceTrigger = voiceTrigger
        self.expansion = expansion
        self.isEnabled = isEnabled
    }
}

File: AIDictationAndroid/app/src/main/java/com/whispermate/aidictation/domain/model/Shortcut.kt
package com.whispermate.aidictation.domain.model

import com.squareup.moshi.JsonClass
import java.util.UUID

@JsonClass(generateAdapter = true)
data class Shortcut(
    val id: String = UUID.randomUUID().toString(),
    val voiceTrigger: String,
    val expansion: String,
    val isEnabled: Boolean = true
)
